sw/device/lib/dif/dif_edn.h

```diff
@@ -0,0 +1,384 @@
+   */
+  uint32_t data[12];
+} dif_edn_seed_material_t;
+
+/**
+ * Auto-generate EDN module configuration parameters.
+ */
+typedef struct dif_edn_auto_params {
+  /**
+   * CSRNG instantiate command parameters.
+   */
+  dif_edn_seed_material_t instantiate_params;
+  /**
+   * CSRNG reseed command parameters.
+   */
+  dif_edn_seed_material_t reseed_params;
+  /**
+   * Number of uint32_t words to request the CSRNG on each generate call rounded
+   * to the nearest 128bit block.
+   */
+  uint32_t generate_len;
+  /**
+   * Number of generate calls that can be made before a reseed request is made.
+   */
+  uint32_t reseed_interval;
+} dif_edn_auto_params_t;
+
+/**
+ * A Entropy Distribution Network interrupt request type.
+ */
+typedef enum dif_edn_irq {
+  /**
+   * Asserted when a CSRNG request has completed.
+   */
+  kDifEdnCmdReqDone,
+  /**
+   * Asserted when a FIFO error occurs.
+   */
+  kDifEdnFatalError,
+} dif_edn_irq_t;
+
+/**
+ * A snapshot of the enablement state of the interrupts for Entropy Distribution
+ * Network.
+ *
+ * This is an opaque type, to be used with the `dif_edn_irq_disable_all()` and
+ * `dif_edn_irq_restore_all()` functions.
+ */
+typedef uint32_t dif_edn_irq_snapshot_t;
+
+/**
+ * Creates a new handle for Entropy Distribution Network.
+ *
+ * This function does not actuate the hardware.
+ *
+ * @param params Hardware instantiation parameters.
+ * @param[out] edn Out param for the initialized handle.
+ * @return The result of the operation.
+ */
+DIF_WARN_UNUSED_RESULT
+dif_edn_result_t dif_edn_init(dif_edn_params_t params, dif_edn_t *edn);
+
+/**
+ * Configures Entropy Distribution Network with runtime information.
+ *
+ * This function should need to be called once for the lifetime of `handle`.
+ *
+ * @param edn A Entropy Distribution Network handle.
+ * @return The result of the operation.
+ */
+DIF_WARN_UNUSED_RESULT
+dif_edn_result_t dif_edn_configure(const dif_edn_t *edn);
+
+/**
+ * Enables the Entropy Distribution Network in boot-time mode.
+ *
+ * Each call to this function should be sequenced with a call to
+ * `dif_edn_stop()`.
+ *
+ * @param edn A Entropy Distribution Network handle.
+ * @return The result of the operation.
+ */
+DIF_WARN_UNUSED_RESULT
+dif_edn_result_t dif_edn_boot_mode_start(const dif_edn_t *edn);
+
+/**
+ * Enables the Entropy Distribution Network in auto refresh mode.
+ *
+ * Each call to this function should be sequenced with a call to
+ * `dif_edn_stop()`.
+ *
+ * @param edn A Entropy Distribution Network handle.
+ * @param config Auto request configuration parameters.
+ * @return The result of the operation.
+ */
+DIF_WARN_UNUSED_RESULT
+dif_edn_result_t dif_edn_auto_mode_start(const dif_edn_t *edn,
+                                         dif_edn_auto_params_t *config);
+
+/**
+ * EDN Status flags.
+ */
+typedef enum dif_edn_status {
+  /**
+   * Device is actively processing a command in either auto or boot-time mode of
+   * operation.
+   */
+  kDifEdnStatusCmdStatus,
```

Comment:
I think `CmdRunning` or something like that might be more accurate?

Reply:
Agreed and Done